app/services/market_order_dedupe_service.rb

```diff
@@ -57,6 +57,8 @@ def dedupe
         if REDIS[@server_id].get("RECORD_SHA256:#{sha256}").nil?
           REDIS[@server_id].set("RECORD_SHA256:#{sha256}", '1', ex: 600)
 
+          next if order['LocationId'] == 0 || !order['LocationId'].is_a?(Numeric)  # sometimes the client sends us 0 for the numeric location id, or a string, we trash this data
```

Comment:
there should be tests for both cases for these

Reply:
Done